index.js

```diff
@@ -182,8 +181,17 @@ exports.fromComment = function (comment) {
   return new Converter(comment, { encoding: encoding, hasComment: true });
 };
 
-exports.fromMapFileComment = function (comment, dir) {
-  return new Converter(comment, { commentFileDir: dir, isFileComment: true, isJSON: true });
+function makeConverter(sm) {
+  return new Converter(sm, { isJSON: true });
+}
+
+exports.fromMapFileComment = function (comment, read) {
```

Comment:
Wow this breaks compatibility a ton.

Given this is used mostly in Node.js I wish there was a way to still make the old behavior available somehow.
This suddenly became way more complex, i.e. before all I had to do is provide the `dir` and now I need to figure all this out by hand, reading the file and such.

Couldn't we keep the old behavior for Node.js (checking if `dir|read` is a `string` or a `function` (printing a deprecation message if it is a `string` with instructions to upgrade).
This could work in conjunction with the [browser field](https://github.com/defunctzombie/package-browser-field-spec) in order to avoid having to import `fs` might be a better solution.

Reply:
Instead of the original behavior. I was thinking we could do this:
```js
if (typeof read === 'string') {
   throw new Error('String directories are no longer supported. Please pass the following function: function (filename) { return fs.readFileSync(path.resolve("' + read '", filename), "utf-8") }'
}
```